pallets/msa/src/tests.rs

```diff
@@ -1640,6 +1641,107 @@ fn replaying_add_provider_to_msa_fails() {
 	})
 }
 
+// Assert that check nonce validation does not create a token account for delete_msa_key call.
+#[test]
+fn signed_ext_check_nonce_delete_msa_key() {
+	new_test_ext().execute_with(|| {
+		let (key_pair, _) = sr25519::Pair::generate();
+		let new_key = key_pair.public();
+
+		let len = 0_usize;
+
+		let call_delete_msa_key: &<Test as frame_system::Config>::Call =
+			&Call::Msa(MsaCall::delete_msa_key { key: AccountId32::from(new_key) });
+		let info = call_delete_msa_key.get_dispatch_info();
+
+		let who = test_public(1);
+
+		assert_ok!(CheckNonce::<Test>(0).pre_dispatch(&who, call_delete_msa_key, &info, len));
+
+		let created_token_account: bool;
+		match frame_system::Account::<Test>::try_get(who) {
+			Ok(_) => {
+				created_token_account = true;
+			},
+			Err(_) => {
+				created_token_account = false;
+			},
+		};
+		assert_eq!(created_token_account, false);
+	})
+}
+
+// Assert that check nonce validation does not create a token account for revoke_msa_delegation_by_delegator call.
+#[test]
+fn signed_ext_check_nonce_revoke_msa_delegation_by_delegator() {
+	new_test_ext().execute_with(|| {
+		let (provider_msa_id, _) = test_create_delegator_msa_with_provider();
+
+		// We are testing the revoke_msa_delegation_by_delegator call.
+		let call_revoke_msa_delegation_by_delegator: &<Test as frame_system::Config>::Call =
+			&Call::Msa(MsaCall::revoke_msa_delegation_by_delegator { provider_msa_id });
+
+		let len = 0_usize;
+
+		// Get the dispatch info for the call.
+		let info = call_revoke_msa_delegation_by_delegator.get_dispatch_info();
+
+		// 1.  Test that revoke_msa_delegation_by_delegator() does not generate token account
+		let who = test_public(1);
+
+		assert_ok!(CheckNonce::<Test>(0).pre_dispatch(
+			&who,
+			call_revoke_msa_delegation_by_delegator,
+			&info,
+			len
+		));
+
+		let created_token_account: bool;
+		match frame_system::Account::<Test>::try_get(who) {
+			Ok(_) => {
+				created_token_account = true;
+			},
+			Err(_) => {
+				created_token_account = false;
+			},
+		};
+		assert_eq!(created_token_account, false);
+	})
+}
+
+// Assert that check nonce validation does create a token account for a paying call.
+#[test]
+fn signed_ext_check_nonce_creates_token_account_if_paying() {
```

Comment:
One other to do it is to write the tests in check nonce file instead of in this specific pallet just like https://github.com/paritytech/substrate/blob/master/frame/system/src/extensions/check_nonce.rs#L126-L167
One other advantage is that you won't need to add that extra dependency

Reply:
Despite the implementation being broader, the unit tests are actually specific to the MSA pallet as the issue cites two specific MSA functions that need this special override.